File: src/unit-test-data/Data.Tests/Tests.DataTest/Int32/Test.Int32.cs
using Xunit;

namespace PrimeFuncPack.UnitTest.Data.Tests;

partial class TestDataTest
{
    [Theory]
    [InlineData(TestData.MinusFifteen, -15)]
    [InlineData(TestData.MinusOne, -1)]
    [InlineData(TestData.Zero, 0)]
    [InlineData(TestData.One, 1)]
    [InlineData(TestData.PlusFifteen, 15)]
    public void Int32_ExpectTheValue(int actual, int expected)
        =>
        Assert.Equal(expected, actual);
}

File: src/UnitTest.Data.Tests/Tests.DataTest/Test.Int32.fs
namespace PrimeFuncPack.UnitTest.Data.Tests

open Xunit
open PrimeFuncPack.UnitTest
open TestHelper

module Int32Test =

    [<Theory>]
    [<InlineData(TestData.MinusFifteen, -15)>]
    [<InlineData(TestData.MinusOne, -1)>]
    [<InlineData(TestData.Zero, 0)>]
    [<InlineData(TestData.One, 1)>]
    [<InlineData(TestData.PlusFifteen, 15)>]
    let ``Int32 test data value sould be expected`` (testDataValue: int) expected =
        testDataValue |> shouldBeEqual expected